### PR TITLE
[Backport to release/10.1] [Payments NOX] Prevent activate payments dead-ends and allow NOX profile reset

### PR DESCRIPTION
**Pull request by woocommercebot**
This PR is a cherry-pick of #60319 and #60192 to `release/10.1`.

I cherry-picked the #60192 also because there are too many changes there that are dependencies for #60319. Luckly, both PRs tackle things in the same area so it is appropriate that they go together. The testing instructions cover both PRs. 

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

- We reduce our NOX in-context business verification step completion logic reliance on the stored step status as a means to prevent weird dead-ends where the user has no account but can't activate payments.
- To further allow escape-hatches, we provide a WooPayments contextual menu "Reset onboarding" link when no account is connected to force-reset the NOX in-context stored progress.
- We also resolved some edge-cases where the NOX step progression ended up with a blocked test account step that confused the user. Now the test account step is force-marked as completed whenever we initialize a live account.

Closes WOOPLUG-5298
Contributes to WOOPMNT-5253

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Unit tests should pass.
2. Checkout this PR's branch on your local WooCommerce installation, the one that is used together with our local WooPayments installation.
3. Build WooCommerce by running `cd ./plugins/woocommerce && npm run build`
4. Go to your local WooPayments installation and delete/reset any account you may have connected.
5. Make sure you have dev mode enable via the WCPay Dev Tools.
6. Make sure you have the [Options View](https://wordpress.org/plugins/options-view/) plugin installed and activated. 
7. Go to Tools > Options view and delete the `woocommerce_woopayments_nox_profile` DB option.
8. Go to WooCommerce > Settings > Payments, click on "Complete setup" for WooPayments, complete the "Choose your payment methods" step, and close the modal.
9. In the contextual menu for WooPayments you should have a "Reset onboarding" link:
Screenshot 2025-08-12 at 13 57 01

10. Click "Reset onboarding", and you should see a confirmation modal with copy tailored for this scenario:
Screenshot 2025-08-12 at 13 58 03

11. Click "Yes, reset onboarding" and the providers should refresh.
12. You should NOT have a "Reset onboarding" item in the contextual menu for WooPayments.
13. Click on "Complete setup" for WooPayments, complete the "Choose your payment methods" step, and close the modal.
14. Click again on "Complete setup" for WooPayments and you should be presented directly with the "Activate payments" step in the state where you can choose to activate or test payments:
Screenshot 2025-08-12 at 14 00 15

15. Click on "Test payments" and wait until you have a test account onboarded:
Screenshot 2025-08-12 at 14 01 43

16. Click on the "Set up payments" task and you should be redirected to the Payments Settings page.
17. Go to your local Transact Platform [PHPMyAdmin](http://localhost:8087/index.php?route=/sql&pos=0&db=wordpress&table=wcpay_accounts) and mark the current account as deleted.
18. Go to WCPay Dev Tools and clear the account cache. You should have an empty array in the cache.
19. Go to WooCommerce > Settings > Payments and WooPayments should feature a "Complete setup" button and a "Reset onboarding" link in the contextual menu:
Screenshot 2025-08-12 at 14 04 43

20. Click "Complete setup" and notice that the payment methods step is marked as completed and you are presented with the business verification step since the test account is already stored as completed in the NOX profile.
21. Fill in the business verification self-assessment sub-step but don't click on "Continue"; just close the modal.
Screenshot 2025-08-12 at 14 08 02

22. Click "Complete setup" and you should be presented with the business verification step with the values you previously selected.
23. Close the modal and click "Reset onboarding" for WooPayments.
24. Click "Complete setup" and you should be presented with the payment methods step.
25. Complete the payment methods step, choose "Start accepting payments", complete the self-assessment, and you should be presented with the embedded KYC:
Screenshot 2025-08-12 at 14 11 29

26. Complete the 2FA embedded step and close the modal.
27. Click on "Complete setup" and you should be presented with the embedded KYC step where you dropped off:
Screenshot 2025-08-12 at 14 12 31

28. Go to your local Transact Platform [PHPMyAdmin](http://localhost:8087/index.php?route=/sql&pos=0&db=wordpress&table=wcpay_accounts) and mark the current account as deleted.
29. Go to WCPay Dev Tools and clear the account cache. You should have an empty array in the cache.
30. Go to WooCommerce > Settings > Payments and WooPayments should feature a "Complete setup" button and a "Reset onboarding" link in the contextual menu.
31. Click on "Continue setup" and you should be presented with the Activate payments step with the self-assessment data already prefilled (since it is stored in the NOX profile):
Screenshot 2025-08-12 at 14 15 00

32. Close the modal, click on "Reset onboarding", confirm it, and start the onboarding again.
33. Notice that you are taken back to the payment methods set.
34. Complete the payment methods step and you should be presented with the choice between live and test account.
35. Choose to activate payments and complete the onboarding flow and click on "Close this window" at the end.
36. Go to WooCommerce > Settings > Payments and click on "Reset account" for WooPayments. It should work and you should be able to restart the NOX flow from scratch.

### Testing that has already taken place:

The extensive testing and battle testing done in the original PR.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by mordeth — APPROVED** [review]
Battle-tested onboarding, onboarding reset and account reset from Settings > Payments, LYS and Settings > Payments and everything works as expected.

**Review by vladolaru — APPROVED** [review]
I've battle-tested the changes and everything checks out.

**Comment by vladolaru** [comment]
Did another round of testing and things are in good working order!

**Comment by mordeth** [comment]
I did another round of testing with the latest changes, and everything worked as expected.

**Comment by vladolaru** [comment]
@coderabbitai full review

File: plugins/woocommerce/client/admin/client/settings-payments/components/buttons/activate-payments-button.tsx [hunk 1]
```diff
@@ -77,8 +82,28 @@ export const ActivatePaymentsButton = ( {
...
+						from: 'activate_payments_button',
+						source: wooPaymentsOnboardingSessionEntrySettings,
+					}
+				);
+				setOnboardingModalOpen( true );
+			} else {
+				window.location.href = getWooPaymentsSetupLiveAccountLink();
+			}
+		}
+
```
[the remark this answers is not in the document]
Comment (vladolaru) [reply plugins/woocommerce/client/admin/client/settings-payments/components/buttons/activate-payments-button.tsx]: Good catch.

File: plugins/woocommerce/src/Internal/Admin/Settings/Payments.php [hunk 2]
```diff
@@ -72,7 +72,7 @@ final public function init( PaymentsProviders $payment_providers, ExtensionSugge
 	 * @throws Exception If there are malformed or invalid suggestions.
 	 */
 	public function get_payment_providers( string $location, bool $for_display = true ): array {
-		$payment_gateways = $this->providers->get_payment_gateways( $for_display );
+		$payment_gateways = $this->providers->get_payment_gateways( $for_display, $location );
```
[the remark this answers is not in the document]
Comment (vladolaru) [reply plugins/woocommerce/src/Internal/Admin/Settings/Payments.php:75]: Good call.

File: plugins/woocommerce/client/admin/client/settings-payments/components/modals/woo-payments-reset-account-modal.tsx [hunk 3]
```diff
@@ -64,12 +74,14 @@ export const WooPaymentsResetAccountModal = ( {
...
 	 */
 	const handleResetAccount = () => {
 		setIsResettingAccount( true );
 
-		resetWooPaymentsAccount()
+		apiFetch( {
+			url: resetUrl,
+			method: 'POST',
+		} )
 			.then( () => {
```
[the remark this answers is not in the document]
Comment (vladolaru) [reply plugins/woocommerce/client/admin/client/settings-payments/components/modals/woo-payments-reset-account-modal.tsx:104]: Good call.

**Review by elazzabi — APPROVED** [review]
Tested this on JN, works as expected! I tested all cases presented in the PR testing instructions
CleanShot 2025-08-15 at 15 14 50@2x
